pylint_exit_options.py

```diff
@@ -1,269 +1,302 @@
+            1
+            >>> ExitCodeMutator.handle_exit_code(12)
+            The following messages were raised:
+            <BLANKLINE>
+              - warning message issued
+              - refactor message issued
+            <BLANKLINE>
+            No fatal messages detected.  Exiting gracefully...
+            0
+        """
+        messages = self._get_messages(value)
+        exit_code = self._get_exit_code(value)
+
+        if messages:
+            print('The following types of issues were found:')
+            print('')
+
+            for message in messages:
+                print("  - %s" % message)
+
+            print('')
+
+        if exit_code:
+            print('The following types of issues are blocking:')
+            print('')
+
+            exit_messages = self._get_messages(exit_code)
+            for exit_message in exit_messages:
+                print("  - %s" % exit_message)
+
+            print('')
+
+        return exit_code
+
+    def _handle_cli_arg(self, namespace):
+        """Applies the CLI flags
+
+        Args:
+            namespace (argparse.Namespace): namespace from CLI arguments
+
+        """
+        if namespace.exit_report:
+            arg_value_list = namespace.exit_report.split(',')
+            self._set_report_arg_values(arg_value_list)
+
+    def _set_report_arg_values(self, arg_values: []):
+        """ Apply an enforcement setting
+
+        Args:
+            arg_values (List): A list of setting passed into the '--exit-report' option which can change which exit
+            codes will be returned to cli
+        """
+        if 'F' in arg_values:
+            self._apply_enforcement_setting(__FATAL__, __FATAL__)
+        else:
+            self._apply_enforcement_setting(__FATAL__, __SUPPRESS__)
+        if 'E' in arg_values:
+            self._apply_enforcement_setting(__ERROR__, __ERROR__)
+        else:
+            self._apply_enforcement_setting(__ERROR__, __SUPPRESS__)
+        if 'W' in arg_values:
+            self._apply_enforcement_setting(__WARNING__, __WARNING__)
+        else:
+            self._apply_enforcement_setting(__WARNING__, __SUPPRESS__)
+        if 'R' in arg_values:
+            self._apply_enforcement_setting(__REFACTOR__, __REFACTOR__)
+        else:
+            self._apply_enforcement_setting(__REFACTOR__, __SUPPRESS__)
+        if 'C' in arg_values:
+            self._apply_enforcement_setting(__CONVENTION__, __CONVENTION__)
+        else:
+            self._apply_enforcement_setting(__CONVENTION__, __SUPPRESS__)
+        if 'U' in arg_values:
+            self._apply_enforcement_setting(__USAGE__, __USAGE__)
+        else:
+            self._apply_enforcement_setting(__USAGE__, __SUPPRESS__)
+
+    def _apply_enforcement_setting(self, key, value):
+        """ Apply an enforcement setting
+
+        Args:
+            key (int): specific message level to set
+            value (int): new value for level
+
+        """
+        positions = {
```

Comment:
Perhaps this could be a class variable so that you don't re-initialize it every time the method is called?  Not sure if Python handles these sorts of things differently, though.

Reply:
The position array is completely gone now